internal/pam_issuer_server/server.go

```diff
@@ -45,7 +48,11 @@ func New(
 }
 
 func oapiErrorHandler(w http.ResponseWriter, message string, statusCode int) {
-	http.Error(w, fmt.Sprintf("API Error: %s", message), statusCode)
+	if statusCode < 0 || statusCode > math.MaxInt32 {
+		statusCode = http.StatusInternalServerError
+	}
+	status := api.NewFailureStatus(int32(statusCode), http.StatusText(statusCode), fmt.Sprintf("API Error: %s", message)) // #nosec G115 -- statusCode validated above
+	transport.SetResponse(w, nil, status)
```

Comment:
in any case this shouldnt have effect - per OIDC spec errors are returned in a certain way